lib/std/atomic.zig

```diff
@@ -402,6 +414,22 @@ pub inline fn spinLoopHint() void {
                 asm volatile ("" ::: "memory");
             }
         },
+
+        // The 8-bit immediate specifies the amount of cycles to pause for. We can't really be too
+        // opinionated here.
+        .hexagon,
+        => asm volatile ("pause(#1)" ::: "memory"),
```

Comment:
I'm a bit on the fence with this one. Unlike the others, it is not merely a hint, but actually *will* suspend execution for a number of cycles (1 in this case). It is still explicitly mentioned as appropriate for spin locks, however. See §8.9 here: https://developer.qualcomm.com/qfile/70668/80-n2040-53_aa_prm_v73.pdf

Reply:
I think this is a valid lowering of the documented semantics ("Signals to the processor that the caller is inside a busy-wait spin-loop.") :+1: